config/cdc/postgres-entrypoint.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+# This script replaces the default docker entrypoint for postgres in the
+# development environment.
+# Its job is to ensure postgres is properly configured to support the
+# Change Data Capture pipeline (by setting access permissions and installing
+# the replication plugin we use for CDC). Unfortunately the default
+# Postgres image does not allow this level of configurability so we need
+# to do it this way in order not to have to publish and maintain our own
+# Postgres image.
+#
+# This then, at the end, transfers control to the default entrypoint.
+
+set -e
+
+cdc_setup_hba_conf() {
+    # Ensure pg-hba is properly configured to allow connections
+    # to the replication slots.
+
+    PG_HBA="$PGDATA/pg_hba.conf"
+    if [ ! -f "$PG_HBA" ]; then
+        echo "DB not initialized. Postgres will take care of pg_hba"
+    fi
+
+    if [ "$(grep -c -E "^host\s+replication" "$PGDATA"/pg_hba.conf)" -ge 0 ]; then
+        echo "Replication config already present in pg_hba. Not changing anything."
+    else
+        echo "host replication all all $POSTGRES_HOST_AUTH_METHOD" >> "$PGDATA/pg_hba.conf"
+    fi
+}
+
+install_wal2json() {
+    # Install the latest version of wal2json if it is not already
+    # present in /wal2json
+    # If we cannot download the latest version from github the following
+    # attempts are made:
+    # - see if there is a valid version on the volume. Use that
+    # - see if for any reason there is already a version in the postgres
+    #   lib directory
+    # If not it is a bad day. And this stops the process.
+
+    set +e
+
+    ARCH=$(uname -m)
+    FILE_NAME="wal2json-Linux-$ARCH.so"
+    LATEST_VERSION=$(
+        wget "https://api.github.com/repos/getsentry/wal2json/releases/latest" -O - |
+        grep '"tag_name":' |
+        sed -E 's/.*"([^"]+)".*/\1/'
+    )
+
+    if [[ $LATEST_VERSION ]]; then
+        if [ ! -f "/wal2json/$LATEST_VERSION/$FILE_NAME" ]; then
+            mkdir -p "/wal2json/$LATEST_VERSION"
+            if wget \
+                "https://github.com/getsentry/wal2json/releases/download/$LATEST_VERSION/$FILE_NAME" \
+                -P "/wal2json/$LATEST_VERSION/"; then
+                echo "$LATEST_VERSION" > /wal2json/latest_version
+            fi
+        fi
+        cp "/wal2json/$LATEST_VERSION/$FILE_NAME" /usr/local/lib/postgresql/wal2json.so
+    elif [ -f /wal2json/latest_version ]; then
```

Comment:
Sort of a crazy idea but we may consider embedding the version information into the binary itself and extract it from there, making this bookkeeping unnecessary.

A somewhat crazy example: https://stackoverflow.com/a/3314891/90297

Reply:
If we wanted to, we could also include the version within the filename released.

Reply:
Are you sure extracting information from the binary would be a safer idea than boookkeeping? I doubt that would translate easily to other architectures.

Reply:
> If we wanted to, we could also include the version within the filename released.

I was thinking about this actually, but I am not sure we can have access to the version when we generate the release. Though We do not need to add the version id in the published release. We just need to attach it to the filename when we store it here. 
Will try this way.

Reply:
Was just a suggestion. Your concern re architecture sounds legit to me. I just don't like not being able to get the binary version from itself directly.

Reply:
I removed the bookkeeping file. Now that is replaced by a symbolic link to the actual library file. So we cannot have an inconsistency anymore between a bookkeeping file and the actual library.